lib/bolt/transport/winrm/connection.rb

```diff
@@ -140,12 +142,59 @@ def mkdirs(dirs)
         end
 
         def write_remote_file(source, destination)
+          if target.options['file-protocol'] == 'smb'
+            write_remote_file_smb(source, destination)
+          else
+            write_remote_file_winrm(source, destination)
+          end
+        end
```

Comment:
Looking for feedback about the following questions:

* What's the best way to select smb over winrm for file copy operations? Should it be a setting in the `winrm` section? Should it be part of the protocol, eg `winrm+smb://`? If so, can it be configured globally?
* It'd be nice to configure some aspects of SMB, such whether to disallow SMBv1, whether to require SMB signing, and in the future options for SMBv3. Should that be a subsection within `winrm`? Or should it be at the same level as `winrm`, `ssh`, sections?
* Does bolt support `sftp` and `scp` for file transfers?

Reply:
part of the protocol would make this a separate transport which I don't think is ideal. I think it's best for this to just be part of the `winrm` transport enabled with a config option. If there are `smb` specific options those should just be more `winrm` configuration options. If you expect there to be more than a few they can be nested inside a `smb` key